plugins/cilium-cni/cilium-cni.go

```diff
@@ -599,6 +649,18 @@ func cmdDel(args *skel.CmdArgs) error {
 		return fmt.Errorf("unable to connect to Cilium daemon: %s", client.Hint(err))
 	}
 
+	// Retrieve configuration so we can tell whether delegated IPAM is enabled or not.
+	configResult, err := c.ConfigGet()
+	if err != nil {
+		return fmt.Errorf("unable to retrieve configuration from cilium-agent: %s", err)
+	}
+
+	if configResult == nil || configResult.Status == nil {
+		return fmt.Errorf("did not receive configuration from cilium-agent")
+	}
+
+	conf := *configResult.Status
```

Comment:
Is there a reason we need this in the delete but not in the add case? I would have thought that the CNI binary still has access to the configuration object to be able to detect that delegated IPAM is in use, even from a `CNI DEL` call.

Reply:
yep, the ADD command already had a call to `c.ConfigGet()` to read `conf.DatapathMode`